src/core/CMakeLists.txt

```
@@ -215,6 +215,12 @@ target_include_directories(openthread-ftd PUBLIC ${OT_PUBLIC_INCLUDES} PRIVATE $
 target_include_directories(openthread-mtd PUBLIC ${OT_PUBLIC_INCLUDES} PRIVATE ${COMMON_INCLUDES})
 target_include_directories(openthread-radio PUBLIC ${OT_PUBLIC_INCLUDES} PRIVATE ${COMMON_INCLUDES})
 
+
+set(OT_EXTENSION_SOURCE "" CACHE STRING "set extension source for OpenThread")
+if(OT_EXTENSION_SOURCE)
+  target_compile_definitions(ot-config INTERFACE "OPENTHREAD_ENABLE_VENDOR_EXTENSION=1")
+  list(APPEND COMMON_SOURCES ${PROJECT_SOURCE_DIR}/${OT_EXTENSION_SOURCE})
+endif()
```

The source path can be an absolute path.
```suggestion
set(OT_VENDOR_EXTENSION "" CACHE STRING "specify a C++ source file built as part of OpenThread core library")
if(OT_VENDOR_EXTENSION)
  target_compile_definitions(ot-config INTERFACE "OPENTHREAD_ENABLE_VENDOR_EXTENSION=1")
  list(APPEND COMMON_SOURCES ${OT_VENDOR_EXTENSION})
endif()
```